### PR TITLE
fix caching test failure after replacing MULTI with AUTO cumulative throughput mode.

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Disable reading model when compiling model to AUTO with cumulative throughput mode.
 - Update logic for AUTO compiling model with cumulative throughput mode.
 

### Tickets:
 - 104939

### CODE REVIEW

File: src/plugins/auto/auto_schedule.cpp [hunk 1]
```diff
@@ -234,20 +234,14 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
     bool isCumulative =
         (_autoSContext->_performanceHint == IE::PluginConfigParams::CUMULATIVE_THROUGHPUT) ? true : false;
     if (isCumulative) {
-        std::list<DeviceInformation> validDevices =
-            _autoSContext->_plugin->GetValidDevice(_autoSContext->_devicePriorities,
-                                                   _loadContext[ACTUALDEVICE].networkPrecision);
```
Comment (songbell) [inline src/plugins/auto/auto_schedule.cpp:239]: why not call getvaliddevice

File: src/plugins/auto/auto_schedule.cpp [hunk 2]
```diff
@@ -527,7 +521,7 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
             _passthroughExeNet = _loadContext[ACTUALDEVICE].executableNetwork;
         }
     }
-    WaitFirstNetworkReady();
+    _autoSContext->_hwExecutableNetwork = WaitFirstNetworkReady();
 }
```
Comment (songbell) [inline src/plugins/auto/auto_schedule.cpp:525]: can we simplify the implementation with _firstLoadOC @yangwang201911  @wgzintel

File: src/plugins/auto/auto_schedule.cpp (hunk 1 above)
Comment (yangwang201911) [reply src/plugins/auto/auto_schedule.cpp:239]: will get empty list after calling this function if the inputting priority list is "mock.o,mock.1,......".

Comment (songbell) [reply src/plugins/auto/auto_schedule.cpp:239]: why,auto is using black list, is mock in black list?

Comment (yangwang201911) [reply src/plugins/auto/auto_schedule.cpp:239]: No. It's not caused by black list. From my understand, function GetVaildDevice() just parses out the devices based on selection priority like  "dGPU > VPUX > iGPU > MYRIAD > CPU", which didn't include the mock device.

**Pull request merged**